pkg/controllers/allocation/bind.go

```diff
@@ -59,10 +65,41 @@ func (b *Binder) Bind(ctx context.Context, node *v1.Node, pods []*v1.Pod) error
 		if !errors.IsAlreadyExists(err) {
 			return fmt.Errorf("creating node %s, %w", node.Name, err)
 		}
+		// If the node object already exists, make sure finalizer and taint are in place.
+		if err := b.KubeClient.Patch(ctx, node, client.StrategicMergeFrom(stored)); err != nil {
```

Comment:
We have a controller (node/finalizer.go) that enforces this already. Further, if we move forward with this change, we should make sure that ```v1.Taint{
		Key:    v1alpha3.NotReadyTaintKey,
		Effect: v1.TaintEffectNoSchedule,
	}``` is applied as part of the user data.

Reply:
OK, I removed the `Patch` call and `v1alpha3.NotReadyTaintKey` is now appended to the `ReadinessTaints` to make sure it gets added in the user data (launch template).